File: frontend/types/design-request.ts
import type { SourceSystem } from "./brief";

export type RequestPriority = "low" | "medium" | "high" | "urgent";

export type DesignRequestStatus =
  | "draft"
  | "submitted"
  | "triaged"
  | "queued"
  | "in_production"
  | "in_internal_review"
  | "in_stakeholder_approval"
  | "changes_requested"
  | "approved"
  | "delivered"
  | "archived"
  | "cancelled";

export interface DesignRequest {
  id: string;
  brief_id: string;
  project_id: string;
  brand_id: string;
  campaign_id?: string | null;
  source_system: SourceSystem;
  source_reference_id?: string | null;
  request_type: string;
  priority: RequestPriority;
  status: DesignRequestStatus;
  current_stage: string;
  owner_team?: string | null;
  assigned_lead_name?: string | null;
  risk_level?: string | null;
  sla_due_at?: string | null;
  round_number: number;
  requested_at: string;
  approved_at?: string | null;
  delivered_at?: string | null;
  created_at: string;
  updated_at: string;
}

export interface DesignRequestCreateInput {
  brief_id: string;
  project_id: string;
  brand_id: string;
  campaign_id?: string | null;
  source_system: SourceSystem;
  source_reference_id?: string | null;
  request_type: string;
  priority: RequestPriority;
  status: DesignRequestStatus;
  current_stage: string;
  owner_team?: string | null;
  assigned_lead_name?: string | null;
  risk_level?: string | null;
  sla_due_at?: string | null;
}

export interface DesignRequestUpdateInput {
  priority?: RequestPriority;
  status?: DesignRequestStatus;
  current_stage?: string;
  owner_team?: string | null;
  assigned_lead_name?: string | null;
  risk_level?: string | null;
  sla_due_at?: string | null;
}

export interface DesignRequestListResponse {
  data: DesignRequest[];
  meta: {
    count: number;
    limit: number;
    offset: number;
  };
  error: null;
}

export interface DesignRequestCreateResponse {
  data: DesignRequest;
  meta: {
    count: number;
    [key: string]: number | string;
  };
  error: null;
}

File: backend/app/schemas/design_request.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.models.enums import DesignRequestStatus, RequestPriority, SourceSystem
from app.schemas.common import TimestampedSchema


class DesignRequestBase(BaseModel):
    brief_id: UUID
    project_id: UUID
    brand_id: UUID
    campaign_id: UUID | None = None
    source_system: SourceSystem
    source_reference_id: str | None = None
    request_type: str
    priority: RequestPriority
    status: DesignRequestStatus
    current_stage: str
    owner_team: str | None = None
    assigned_lead_name: str | None = None
    risk_level: str | None = None
    sla_due_at: datetime | None = None


class DesignRequestCreate(DesignRequestBase):
    pass


class DesignRequestUpdate(BaseModel):
    priority: RequestPriority | None = None
    status: DesignRequestStatus | None = None
    current_stage: str | None = None
    owner_team: str | None = None
    assigned_lead_name: str | None = None
    risk_level: str | None = None
    sla_due_at: datetime | None = None


class DesignRequestRead(DesignRequestBase, TimestampedSchema):
    id: UUID
    round_number: int
    requested_at: datetime
    approved_at: datetime | None = None
    delivered_at: datetime | None = None


class DesignRequestListResponse(BaseModel):
    data: list[DesignRequestRead]
    meta: dict[str, int]
    error: None = None


class DesignRequestCreateResponse(BaseModel):
    data: DesignRequestRead
    meta: dict[str, int | str]
    error: None = None
